cmd/internal/cli/build_darwin.go

```diff
@@ -27,7 +27,7 @@ func run(cmd *cobra.Command, args []string) {
 		os.Exit(1)
 	}
 
-	if !remote {
+	if !remote && !cmd.Flags().Lookup("builder").Changed {
```

Comment:
I see what you are doing here: either the use set remote explicitly or they specified builder and therefore set remote implicitly.

In order to reduce maintaining costs and avoid having to check the two things continuously, I think it might be easier to check for builder Changed once, and *set* remote if that's the case. Maybe take preRun out of build_linux.go and build_darwin.go (since it's the same in both) and move it to build.go, and perform this check there:

```
func preRun(...) {
    ...
    if cmd.Flags().Lookup("builder").Changed {
        cmd.Flags().Lookup("remote").Value.Set("true")
    }
}
```

in this way, in the rest of the code you can continue checking only remote, as it is now and you don't have to introduce the explicit handling of the dependency everywhere.

It's also a good idea to update the documentation for the builder flag and add something like "(implies --remote)".